test/domain.middleware.test.ts

```diff
@@ -0,0 +1,55 @@
+import 'mocha';
+import { Request, Response, NextFunction } from 'express';
+import { expect } from 'chai';
+import { createRequest, createResponse } from 'node-mocks-http';
+
+import * as globalStore from '../src';
+import AsyncStoreAdapter from '../src/AsyncStoreAdapter';
+
+type Imiddleware = (req: Request, res: Response, next: NextFunction) => Promise<any> | void;
```

Comment:
You can export the type from here. https://github.com/leapfrogtechnology/async-store/blob/master/src/Middleware.ts#L6